classes/submission/SubmissionDAO.inc.php

```diff
@@ -111,10 +112,104 @@ function _fromRow($row) {
 		$submission->setLanguage($row['language']);
 		$submission->setCommentsToEditor($row['comments_to_ed']);
 
-		$this->getDataObjectSettings('submission_settings', 'submission_id', $submission->getId(), $submission);
+		$this->getSubmissionSettings($submission, $submissionRevision);
 
 		return $submission;
 	}
+	
+	/**
+	 * Get contents of the submission_settings table
+	 * @param $submission Submission object
+	 * @param $submissionRevision int optional
+	 */
+	function getSubmissionSettings($submission, $submissionRevision = null) {
```

Comment:
The `get` prefix suggests it's going to return a value, but it loads it into the `$submission` instead. Suggest calling the function `loadSubmissionSettings` instead?